### ISSUE TITLE
Installation problems

### ISSUE THREAD

**Issue by mavidser** [firefly-iii/firefly-iii#1508]
**Bug description**
I am running Firefly III version 4.7.4.

I created a new container from the :latest tag, and ran those four mentioned commands to initialize the database. Now, opening `/` redirects to `/login` which redirects to `/install` which seems to be stuck on:

> FireflyIII
> installation and upgrade
> Please wait...
> > Waiting to start...

**Steps to reproduce**
Docker install with mysql connection (MariaDB 10.7.3)
Ran these commands:
```
docker exec -it firefly php artisan migrate --seed
docker exec -it firefly php artisan firefly:upgrade-database
docker exec -it firefly php artisan firefly:verify
docker exec -it firefly php artisan passport:install
```

**Extra info**
Command logs:
```

shell $ docker exec -it firefly php artisan migrate --seed
Migration table created successfully.
Migrating: 2016_06_16_000000_create_support_tables
Migrated:  2016_06_16_000000_create_support_tables
Migrating: 2016_06_16_000001_create_users_table
Migrated:  2016_06_16_000001_create_users_table
Migrating: 2016_06_16_000002_create_main_tables
Migrated:  2016_06_16_000002_create_main_tables
Migrating: 2016_08_25_091522_changes_for_3101
Migrated:  2016_08_25_091522_changes_for_3101
Migrating: 2016_09_12_121359_fix_nullables
Migrated:  2016_09_12_121359_fix_nullables
Migrating: 2016_10_09_150037_expand_transactions_table
Migrated:  2016_10_09_150037_expand_transactions_table
Migrating: 2016_10_22_075804_changes_for_v410
Migrated:  2016_10_22_075804_changes_for_v410
Migrating: 2016_11_24_210552_changes_for_v420
Migrated:  2016_11_24_210552_changes_for_v420
Migrating: 2016_12_22_150431_changes_for_v430
Migrated:  2016_12_22_150431_changes_for_v430
Migrating: 2016_12_28_203205_changes_for_v431
Migrated:  2016_12_28_203205_changes_for_v431
Migrating: 2017_04_13_163623_changes_for_v440
Migrated:  2017_04_13_163623_changes_for_v440
Migrating: 2017_06_02_105232_changes_for_v450
Migrated:  2017_06_02_105232_changes_for_v450
Migrating: 2017_08_20_062014_changes_for_v470
Migrated:  2017_08_20_062014_changes_for_v470
Migrating: 2017_11_04_170844_changes_for_v470a
Migrated:  2017_11_04_170844_changes_for_v470a
Migrating: 2018_01_01_000001_create_oauth_auth_codes_table
Migrated:  2018_01_01_000001_create_oauth_auth_codes_table
Migrating: 2018_01_01_000002_create_oauth_access_tokens_table
Migrated:  2018_01_01_000002_create_oauth_access_tokens_table
Migrating: 2018_01_01_000003_create_oauth_refresh_tokens_table
Migrated:  2018_01_01_000003_create_oauth_refresh_tokens_table
Migrating: 2018_01_01_000004_create_oauth_clients_table
Migrated:  2018_01_01_000004_create_oauth_clients_table
Migrating: 2018_01_01_000005_create_oauth_personal_access_clients_table
Migrated:  2018_01_01_000005_create_oauth_personal_access_clients_table
Migrating: 2018_03_19_141348_changes_for_v472
Migrated:  2018_03_19_141348_changes_for_v472
Migrating: 2018_04_07_210913_changes_for_v473
Migrated:  2018_04_07_210913_changes_for_v473
Migrating: 2018_04_29_174524_changes_for_v474
Migrated:  2018_04_29_174524_changes_for_v474
Seeding: AccountTypeSeeder
Seeding: TransactionCurrencySeeder
Seeding: TransactionTypeSeeder
Seeding: PermissionSeeder
Seeding: LinkTypeSeeder
Seeding: ConfigSeeder

shell $ docker exec -it firefly php artisan firefly:upgrade-database
Updating currency information..
Done updating currency information..
Firefly III database is up to date.

shell $ docker exec -it firefly php artisan firefly:verify
No orphaned transactions!
No orphaned journals!
All link types OK!
All access tokens OK!
Amount integrity OK!

shell $ docker exec -it firefly php artisan passport:install
Encryption keys generated successfully.
Personal access client created successfully.
Client ID: 1
Client Secret: redacted
Password grant client created successfully.
Client ID: 2
Client Secret: redacted
```

Standard logs:
```
chown: cannot access '/var/www/firefly-iii/storage/cache': No such file or directory
chmod: cannot access '/var/www/firefly-iii/storage/cache': No such file or directory
# You can leave this on "local". If you change it to production most console commands will ask for extra confirmation.
# Never set it to "testing".
APP_ENV=local

# Set to true if you want to see debug information in error screens.
APP_DEBUG=

# This should be your email address
SITE_OWNER=

# The encryption key for your database and sessions. Keep this very secure.
# If you generate a new one all existing data must be considered LOST.
# Change it to a string of exactly 32 chars or use command `php artisan key:generate` to generate it
APP_KEY=S0m3R@nd0mString0f32Ch@rsEx@ct1y

# APP_URL and TRUSTED_PROXIES are useful when using Docker and/or a reverse proxy.
APP_URL=
TRUSTED_PROXIES=

# The log channel defines where your log entries go to.
LOG_CHANNEL=

# Database credentials. Make sure the database exists. I recommend a dedicated user for Firefly III
# If you use SQLite, set connection to `sqlite` and remove the database, username and password settings.
DB_CONNECTION=mysql
DB_HOST=mysql
DB_PORT=
DB_DATABASE=firefly
DB_USERNAME=firefly
DB_PASSWORD=redacted

# 'daily' is the default logging mode giving you 5 daily rotated log files in /storage/logs/.
# Several other options exist. You can use 'single' for one big fat error log (not recommended).
# Also available are 'syslog' and 'errorlog' which will log to the system itself.
APP_LOG=syslog

# Log level. You can set this from least severe to most severe:
# debug, info, notice, warning, error, critical, alert, emergency
# If you set it to debug your logs will grow large, and fast. If you set it to emergency probably
# nothing will get logged, ever.
APP_LOG_LEVEL=info

# If you're looking for performance improvements, you could install memcached.
CACHE_DRIVER=file
SESSION_DRIVER=file

# Cookie settings. Should not be necessary to change these.
COOKIE_PATH="/"
COOKIE_DOMAIN=
COOKIE_SECURE=false

# If you want Firefly III to mail you, update these settings
MAIL_DRIVER=
MAIL_HOST=
MAIL_PORT=
MAIL_FROM=
MAIL_USERNAME=
MAIL_PASSWORD=
MAIL_ENCRYPTION=

# Firefly III can send you the following messages
SEND_REGISTRATION_MAIL=true
SEND_ERROR_MESSAGE=false

# Set a Mapbox API key here (see mapbox.com) so there might be a map available at various places.
MAPBOX_API_KEY=

# Set a Fixer IO API key here (see https://fixer.io) to enable live currency exchange rates.
# Please note that this will only work for paid fixer.io accounts because they severly limited
# the free API up to the point where you might as well offer nothing.
FIXER_API_KEY=

# If you wish to track your own behavior over Firefly III, set a valid analytics tracker ID here.
ANALYTICS_ID=

# Most parts of the database are encrypted by default, but you can turn this off if you want to.
# This makes it easier to migrate your database. Not that some fields will never be decrypted.
USE_ENCRYPTION=true

# Leave the following configuration vars as is.
# Unless you like to tinker and know what you're doing.
APP_NAME=FireflyIII
BROADCAST_DRIVER=log
QUEUE_DRIVER=sync
REDIS_HOST=127.0.0.1
REDIS_PASSWORD=null
REDIS_PORT=6379
CACHE_PREFIX=firefly
SEARCH_RESULT_LIMIT=50
PUSHER_KEY=
PUSHER_SECRET=
PUSHER_ID=
DEMO_USERNAME=
DEMO_PASSWORD=
IS_DOCKER=true
IS_SANDSTORM=false
IS_HEROKU=false
BUNQ_USE_SANDBOX=false
TZ=
Do not run Composer as root/super user! See https://getcomposer.org/root for details
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
Discovered Package: davejamesmiller/laravel-breadcrumbs
Discovered Package: fideloper/proxy
Discovered Package: laravel/passport
Discovered Package: laravelcollective/html
Discovered Package: pragmarx/google2fa-laravel
Discovered Package: rcrowe/twigbridge
Package manifest generated successfully.
+------------------------------------------------------------------------------+
|                                                                              |
| Thank you for installing Firefly III, v4.7.4!                                |
| This version of Firefly III has a new import routine. See                    |
| http://bit.ly/FF3-new-import for more information.                           |
|                                                                              |
+------------------------------------------------------------------------------+
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.10. Set the 'ServerName' directive globally to suppress this message
AH00558: apache2: Could not reliably determine the server's fully qualified domain name, using 172.17.0.10. Set the 'ServerName' directive globally to suppress this message
[Sun Jun 24 09:17:06.853937 2018] [mpm_prefork:notice] [pid 1] AH00163: Apache/2.4.25 (Debian) PHP/7.1.18 OpenSSL/1.0.2l configured -- resuming normal operations
[Sun Jun 24 09:17:06.854083 2018] [core:notice] [pid 1] AH00094: Command line: 'apache2 -D FOREGROUND'
172.17.0.2 - - [24/Jun/2018:09:24:05 +0000] "GET / HTTP/1.1" 302 1416 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.87 Safari/537.36"
172.17.0.2 - - [24/Jun/2018:09:24:05 +0000] "GET /login HTTP/1.1" 302 1426 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.87 Safari/537.36"
172.17.0.2 - - [24/Jun/2018:09:24:06 +0000] "GET /install HTTP/1.1" 200 1978 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.87 Safari/537.36"
```

storage/logs/ff3-apache2handler-2018-06-24.log:
```
2018-06-24 09:17:57] local.ERROR: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'firefly.configuration' doesn't exist (SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1) {"exception":"[object] (Illuminate\\Database\\QueryException(code: 42S02): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'firefly.configuration' doesn't exist (SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1) at /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php:664, Doctrine\\DBAL\\Driver\\PDOException(code: 42S02): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'firefly.configuration' doesn't exist at /var/www/firefly-iii/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:82, PDOException(code: 42S02): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'firefly.configuration' doesn't exist at /var/www/firefly-iii/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:80)
[stacktrace]
#0 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(624): Illuminate\\Database\\Connection->runQueryCallback('select `id`, `n...', Array, Object(Closure))
#1 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php(333): Illuminate\\Database\\Connection->run('select `id`, `n...', Array, Object(Closure))
#2 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(1909): Illuminate\\Database\\Connection->select('select `id`, `n...', Array, true)
#3 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Query/Builder.php(1894): Illuminate\\Database\\Query\\Builder->runSelect()
#4 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(481): Illuminate\\Database\\Query\\Builder->get(Array)
#5 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Eloquent/Builder.php(465): Illuminate\\Database\\Eloquent\\Builder->getModels(Array)
#6 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Concerns/BuildsQueries.php(77): Illuminate\\Database\\Eloquent\\Builder->get(Array)
#7 /var/www/firefly-iii/app/Support/FireflyConfig.php(65): Illuminate\\Database\\Eloquent\\Builder->first(Array)
#8 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(221): FireflyIII\\Support\\FireflyConfig->get('is_demo_site', false)
#9 /var/www/firefly-iii/app/Http/Controllers/Controller.php(68): Illuminate\\Support\\Facades\\Facade::__callStatic('get', Array)
#10 /var/www/firefly-iii/app/Http/Controllers/HomeController.php(55): FireflyIII\\Http\\Controllers\\Controller->__construct()
#11 [internal function]: FireflyIII\\Http\\Controllers\\HomeController->__construct()
#12 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(800): ReflectionClass->newInstanceArgs(Array)
#13 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(646): Illuminate\\Container\\Container->build('FireflyIII\\\\Http...')
#14 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Container/Container.php(601): Illuminate\\Container\\Container->resolve('FireflyIII\\\\Http...', Array)
#15 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(734): Illuminate\\Container\\Container->make('FireflyIII\\\\Http...', Array)
#16 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Route.php(226): Illuminate\\Foundation\\Application->make('FireflyIII\\\\Http...')
#17 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Route.php(796): Illuminate\\Routing\\Route->getController()
#18 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Route.php(757): Illuminate\\Routing\\Route->controllerMiddleware()
#19 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(672): Illuminate\\Routing\\Route->gatherMiddleware()
#20 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(652): Illuminate\\Routing\\Router->gatherRouteMiddleware(Object(Illuminate\\Routing\\Route))
#21 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(636): Illuminate\\Routing\\Router->runRouteWithinStack(Object(Illuminate\\Routing\\Route), Object(Illuminate\\Http\\Request))
#22 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(602): Illuminate\\Routing\\Router->runRoute(Object(Illuminate\\Http\\Request), Object(Illuminate\\Routing\\Route))
#23 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Router.php(591): Illuminate\\Routing\\Router->dispatchToRoute(Object(Illuminate\\Http\\Request))
#24 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(176): Illuminate\\Routing\\Router->dispatch(Object(Illuminate\\Http\\Request))
#25 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(30): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}(Object(Illuminate\\Http\\Request))
#26 /var/www/firefly-iii/vendor/fideloper/proxy/src/TrustProxies.php(57): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#27 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(151): Fideloper\\Proxy\\TrustProxies->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#28 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#29 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(31): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#30 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(151): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#31 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#32 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(31): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#33 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(151): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#34 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#35 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#36 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(151): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#37 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#38 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php(51): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#39 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(151): Illuminate\\Foundation\\Http\\Middleware\\CheckForMaintenanceMode->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#40 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Routing/Pipeline.php(53): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#41 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(104): Illuminate\\Routing\\Pipeline->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#42 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(151): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#43 /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(116): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter(Object(Illuminate\\Http\\Request))
#44 /var/www/firefly-iii/public/index.php(76): Illuminate\\Foundation\\Http\\Kernel->handle(Object(Illuminate\\Http\\Request))
#45 {main}
"}
[2018-06-24 09:18:35] local.WARNING: The current installed version (1) is older than the required version (4). Redirect to migrate routine.
[2018-06-24 09:24:00] local.WARNING: The current installed version (1) is older than the required version (4). Redirect to migrate routine.
[2018-06-24 09:24:06] local.WARNING: The current installed version (1) is older than the required version (4). Redirect to migrate routine.
```

storage/logs/ff3-cli-2018-06-24.log:
```
[2018-06-24 09:18:17] local.WARNING: No database version entry is present. Database is assumed to be OLD (version 1).
[2018-06-24 09:18:29] local.ERROR: OAuth client generated but no user associated.
[2018-06-24 09:18:29] local.ERROR: OAuth client generated but no user associated.
```

**Comment by JC5** [comment]
```
Base table or view not found: 1146 Table 'firefly.configuration' doesn't exist (SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1) {"exception":"[object] (Illuminate\\Database\\QueryException(code: 42S02): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'firefly.configuration' doesn't exist (SQL: select `id`, `name`, `data` from `configuration` where `name` = is_demo_site and `configuration`.`deleted_at` is null limit 1) at /var/www/firefly-iii/vendor/laravel/framework/src/Illuminate/Database/Connection.php:664, Doctrine\\DBAL\\Driver\\PDOException(code: 42S02): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'firefly.configuration' doesn't exist at /var/www/firefly-iii/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:82, PDOException(code: 42S02): SQLSTATE[42S02]: Base table or view not found: 1146 Table 'firefly.configuration' doesn't exist at /var/www/firefly-iii/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:80)
```

Mmm, it seems the database was not initialized properly. Either the commands didn't take or something else went wrong. I would suggest trying the initialization commands again, and make sure that the environment variables that refer to the database are correct.

**Comment by mavidser** [comment]
The configuration table exists in the database. But I think I found out what went wrong. I'm guessing loading /install sends some trigger to firefly to start installation. I hadn't set the `APP_URL`, so <base href> was set to http, while I loaded over reverse-proxied https, preventing any further requests.

Not sure if this was the problem, but setting APP_URL to https solved this for me. Thanks! Closing this for now.
